vm-virtio/src/transport/pci_device.rs

```diff
@@ -316,6 +341,16 @@ impl PciDevice for VirtioPciDevice {
         self.interrupt_cb = Some(cb);
     }
 
+    fn assign_msix(&mut self, msi_cb: Arc<MsixClosure>) {
+        let msix_config = self.msix_config.clone();
+
+        let cb = Arc::new(Box::new(move |queue: &Queue| {
+            (msi_cb)(msix_config.lock().unwrap().table_entries[queue.msix_vector as usize].clone())
```

Comment:
Is it possible for table_entries overflow here?

Reply:
It shouldn't, since the `queue.msix_vector` index is set by the guest, based on the size of the table that we provided through PCI config space.
And if for any reason this happens, at runtime Rust would panic and terminate the VM.